stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,404 @@
+        super().__init__(env, ids=ids, prefetch_ids=prefetch_ids)
+        self._reset_counters()
+
+    def create_batch(self):
+        self.ensure_one()
+        batch = self._create_batch(raise_if_not_possible=True)
+        action = {
+            "type": "ir.actions.act_window",
+            "name": batch.name,
+            "res_model": "stock.picking.batch",
+            "res_id": batch.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+        return action
+
+    def _reset_counters(self):
+        self._volume_by_partners = defaultdict(lambda: 0)
+        self._device = None
+        self._remaining_weight = 0
+        self._remaining_nbr_picking_lines = 0
+        self._selected_picking_ids = []
+        self._first_picking = None
+        self._remaining_nbr_bins = None
+        self._previous_selected_picking = None
+
+    def _get_picking_order_by(self):
+        self.ensure_one()
+        if self.group_pickings_by_partner:
+            return "priority desc, scheduled_date asc, partner_id desc, id asc"
+        return "priority desc, scheduled_date asc, id asc"
+
+    def _get_picking_domain_common(self, user=None):
+        domain = [
+            ("picking_type_id", "in", self.picking_type_ids.ids),
+            ("state", "in", ("partially_available", "assigned")),
+            ("batch_id", "=", False),
+            ("user_id", "=", user.id if user else False),
+        ]
+        if not user:
+            domain.append(("printed", "=", False))
+        return domain
+
+    def _get_picking_domain_for_device(self, device):
+        return [
+            ("volume", ">=", device.min_volume),
+            ("volume", "<=", device.max_volume),
+            ("weight", "<=", device.max_weight),
+            ("nbr_picking_lines", "<=", self.maximum_number_of_preparation_lines),
+        ]
+
+    def _get_picking_domain_for_additional(self):
+        """Provides the domain expressing the additional constraints to apply to
+        get the next picking to add to the batch.
+        """
+        excluded_ids = self._selected_picking_ids
+        domain = [
+            (
+                "nbr_picking_lines",
+                "<=",
+                self._remaining_nbr_picking_lines,
+            ),
+            ("id", "not in", excluded_ids),
+            ("weight", "<=", self._remaining_weight),
+        ]
+        user = self._first_picking.user_id
+        if user:
+            domain.append(("user_id", "=", user.id))
+        volume_domains = [
+            [
+                ("volume", "<=", self._get_remaining_volume()),
+            ]
+        ]
+        if self.group_pickings_by_partner:
+            previous_picking = self._previous_selected_picking
+            volume_domains.append(
+                [
+                    ("partner_id", "=", previous_picking.partner_id.id),
+                    ("scheduled_date", "=", previous_picking.scheduled_date),
```

Comment:
This looks very specific. Can you add a comment with the intention here?
Shouldn't this be favored compared to the other volume domain? Because you apply a OR between the 2 domains.

Can you also then keep the case with `group_pickings_by_commercial_partner` that is here https://github.com/OCA/wms/blob/14.0/shopfloor_batch_automatic_creation/actions/picking_batch_auto_create.py#L42

Reply:
@jbaudoux The implementation of the UC behind `group_pickings_by_commercial_partner` seems to ignore all the constrains... Do I miss something? In such a case we should introduce a kind of 'batch_mode' field into the wizard with 2 possibilities:
'based on limit' or 'by commercial partner'...

Reply:
@jgrandguillaume Do you remember why 2 users cannot prepare the same commercial partner for the rollercages process?
Can this be changed to always apply limits of the picking device but ensure commercial partners are not mixed in a batch?